core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -726,45 +744,53 @@ public FileInfo getFileInfo(AlluxioURI path, GetStatusContext context)
       throws FileDoesNotExistException, InvalidPathException, AccessControlException, IOException {
     Metrics.GET_FILE_INFO_OPS.inc();
     long opTimeMs = System.currentTimeMillis();
-    LockingScheme lockingScheme =
-        createLockingScheme(path, context.getOptions().getCommonOptions(), LockPattern.READ, true);
     try (RpcContext rpcContext = createRpcContext();
-         LockedInodePath inodePath = mInodeTree
-             .lockInodePath(lockingScheme.getPath(), lockingScheme.getPattern());
-         FileSystemMasterAuditContext auditContext =
-             createAuditContext("getFileInfo", path, null, inodePath.getInodeOrNull())) {
-      try {
-        mPermissionChecker.checkPermission(Mode.Bits.READ, inodePath);
-      } catch (AccessControlException e) {
-        auditContext.setAllowed(false);
-        throw e;
-      }
-      // Possible ufs sync.
-      if (syncMetadata(rpcContext, inodePath, lockingScheme, DescendantType.ONE)) {
+        FileSystemMasterAuditContext auditContext =
+            createAuditContext("getFileInfo", path, null, null)) {
+
+      if (syncMetadata(rpcContext,
```

Comment:
Is the general approach for RPCs to have a sync phase first, then the actual execution of the RPC? (Before it did the sync within the same contexts of the RPC execution).

Reply:
For All RPCs previously the sync phase typically came at the very beginning of the RPC after the initial permission check. This was the old way:

- create RPC, audit context, lock inode path
- check permissions
- perform sync
- perform RPC
- unlock

Because of the new locking requirements, there is a new paradigm of

- create RPC/audit context
- check if sync is required
  - check permissions
  - lock properly for sync
  - do sync
  - unlock from sync
- lock for RPC
- check permissions
- perform RPC
- unlock

The reason permission check is performed twice is because it requires having a locked inode path before doing the check.